src/app/shared/text-overflow/text-overflow-ellipses.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'publish-text-overflow-ellipses',
+  template: `
+    <div [style.max-height]="(numLines * lineHeight) + unit" [style.line-height]="lineHeight + unit">
+      <ng-content></ng-content>
+    </div>
+  `,
+  styleUrls: ['text-overflow-ellipses.component.css']
+})
+
+export class TextOverflowEllipsesComponent {
+  @Input() numLines: number = 3;
+  @Input() lineHeight: number = 1.2;
+  @Input() unit: string = 'em';
```

Comment:
Typically fonts and related items should be sized in ems for responsive layouts because smaller screen sizes should have larger fonts. But the Publish app layouts are pretty well set on a 20px line height at every screen size. These text overflow components will eventually be moved into a styleguide, and ems seemed like a more generic default for that purpose.